src/base/bittorrent/session.cpp

```diff
@@ -1149,7 +1150,18 @@ void Session::initializeNativeSession()
     loadLTSettings(pack);
     lt::session_params sessionParams {pack, {}};
 #ifdef QBT_USES_LIBTORRENT2
-    sessionParams.disk_io_constructor = customDiskIOConstructor;
+    switch (diskIOType())
+    {
+    case DiskIOType::Posix:
+        sessionParams.disk_io_constructor = customPosixDiskIOConstructor;
+        break;
+    case DiskIOType::MMap:
```

Comment:
(optional) I would even swap the order of this and `case DiskIOType::Posix:`.

Reply:
> (optional) I would even swap the order of this and `case DiskIOType::Posix:`.

I'm not very comfortable dealing with someone else's PR, so this can be changed later.